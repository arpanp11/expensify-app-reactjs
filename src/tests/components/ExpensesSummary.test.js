import React from 'react';
import { shallow } from 'enzyme';
import { ExpensesSummary } from '../../components/ExpensesSummary';

test('should correctly render ExpensesSummry with single expense', () => {
  const wrapper = shallow(
    <ExpensesSummary expenseCount={1} expensesTotal={345} />
  );
  expect(wrapper).toMatchSnapshot();
});

test('should correctly render ExpensesSummry with multiple expenses', () => {
  const wrapper = shallow(
    <ExpensesSummary expenseCount={34} expensesTotal={143789245} />
  );
  expect(wrapper).toMatchSnapshot();
});
