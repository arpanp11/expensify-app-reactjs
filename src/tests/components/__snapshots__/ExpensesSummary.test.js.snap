// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should correctly render ExpensesSummry with multiple expenses 1`] = `
<div
  className="page-header"
>
  <div
    className="content-container"
  >
    <h1
      className="page-header__title"
    >
      Viewing 
      <span>
        34
      </span>
       
      expenses
       totalling
       
      <span>
        $1,437,892.45
      </span>
    </h1>
    <div
      className="page-header__actions"
    >
      <Link
        className="button"
        replace={false}
        to="/create"
      >
        Add Expense
      </Link>
    </div>
  </div>
</div>
`;

exports[`should correctly render ExpensesSummry with single expense 1`] = `
<div
  className="page-header"
>
  <div
    className="content-container"
  >
    <h1
      className="page-header__title"
    >
      Viewing 
      <span>
        1
      </span>
       
      expense
       totalling
       
      <span>
        $3.45
      </span>
    </h1>
    <div
      className="page-header__actions"
    >
      <Link
        className="button"
        replace={false}
        to="/create"
      >
        Add Expense
      </Link>
    </div>
  </div>
</div>
`;
